pkg/spdk/engine.go

```diff
@@ -1299,10 +1299,14 @@ func (e *Engine) ReplicaDelete(spdkClient *spdkclient.Client, replicaName, repli
 	}
 
 	// Detaching the corresponding NVMf controller to remote replica
-	controllerName := helperutil.GetNvmeControllerNameFromNamespaceName(e.ReplicaBdevNameMap[replicaName])
-	e.log.Infof("Detaching the corresponding NVMf controller %v during remote replica %s delete", controllerName, replicaName)
-	if _, err := spdkClient.BdevNvmeDetachController(controllerName); err != nil && !jsonrpc.IsJSONRPCRespErrorNoSuchDevice(err) {
-		return errors.Wrapf(err, "failed to detach controller %s for deleting replica %s", controllerName, replicaName)
+	if _, ok := e.ReplicaBdevNameMap[replicaName]; ok {
+		controllerName := helperutil.GetNvmeControllerNameFromNamespaceName(e.ReplicaBdevNameMap[replicaName])
+		if controllerName != "" {
+			e.log.Infof("Detaching the corresponding NVMf controller %v during remote replica %s delete", controllerName, replicaName)
```

Comment:
I was thinking blindly detaching the controller for the replica was better, no matter if there is a record in `e.ReplicaBdevNameMap`. Any reason why we need to add this check?

Reply:
If `controllerName` is empty string, there are lots of annoying error messages in log.

Reply:
@shuo-wu Any concerns?

Reply:
I don't think this will be often triggered. I am more worried about the case that the controller is there but somehow `e.ReplicaBdevNameMap[replicaName]` is empty

Reply:
I see. Removed the commit. Thank you.